### ISSUE TITLE
Visualizing Segmentation Mask

### ISSUE THREAD

**Issue by sawant-nidhish** [Meize0729/RSBuilding#14]
Hello, I am trying to figure out a way to draw a segmentaion mask for a particular image. When I tried to investigate the outputs of the fine-tuned model, I just saw that the output is a classification vector for each pixel. That is, each pixel value is either 1 or 0, depending on whether the pixel is a part of a building or not. Its there a way to output a segmentation mask from this? 

Secondly, I am facing errors when I try to use mmseg to perform inference on the trained model since the models defined on OPEN-CD use mmseg internally. It first says that the FoundationEncoderDecoder model is not registered, and when I try to do so, there is a key error saying that the key pre trained is not defined. Can you please point me in the right direction to generate a segmentation mask for the image as you have done in the paper? I am mostly concerned with building extraction only

**Comment by RoyAn2386** [comment]
Hi there,
1) You can modify the visualize_hook to save your results: (like this: 
[visualization_hook.zip](https://github.com/user-attachments/files/16975189/visualization_hook.zip)

2) in order to use mmseg to read, load model, you can re-register rsbuilding models, transform, ect. before using init_model() or inference():

![image](https://github.com/user-attachments/assets/79575c9f-b977-41bd-811b-a20a2e8d7a5b)

**Comment by sawant-nidhish** [comment]
Thank you for this. It worked

**Comment by jwd222** [comment]
> Hi there,
> 
> 1. You can modify the visualize_hook to save your results: (like this:
>    [visualization_hook.zip](https://github.com/user-attachments/files/16975189/visualization_hook.zip)
> 2. in order to use mmseg to read, load model, you can re-register rsbuilding models, transform, ect. before using init_model() or inference():
> 
> ![image](https://github.com/user-attachments/assets/79575c9f-b977-41bd-811b-a20a2e8d7a5b)

Hi, thank you for providing the code for saving masks. One thing I wanted to understand is the masks saved have 3 channels making different colored masks in the output image file. What is the significance or meaning of each of the channels could you say? I'm guessing they could mean different classes? or maybe TP or FP as seen in the paper. I'm not quite sure.

Regards
